app/main/posts/modify/post-video.directive.js

```diff
@@ -0,0 +1,92 @@
+module.exports = PostVideo;
+
+PostVideo.$inject = [];
+
+function PostVideo() {
+    return {
+        restrict: 'E',
+        scope: {
+            videoUrl: '='
+        },
+        templateUrl: 'templates/main/posts/modify/video.html',
+        controller: PostVideoController
+    };
+}
+
+PostVideoController.$inject = [
+    '$scope',
+    '$sce',
+    'Util'
+];
+
+function PostVideoController(
+    $scope,
+    $sce,
+    Util
+) {
+    $scope.constructIframe = constructIframe;
+    // $scope.getVideoThumbnail = getVideoThumbnail;
+    activate();
+
+    function activate() {
+        // Is this dody? I dunno it seems dodgy
+        $scope.videoUrl = $sce.trustAsResourceUrl($scope.videoUrl);
+        $scope.previewId = $scope.videoId ? $scope.videoId : Util.simpluUUID();
+    }
+
+    function constructIframe(videoUrl) {
+        parseVideo(videoUrl);
+
+        var preview = angular.element(document.getElementById($scope.previewId));
+
+        if ($scope.videoUrl === undefined) {
+            preview.empty();
+        } else {
+            preview.empty();
+            preview.html(createVideo()).wrap('<div class="video_embed-fluid" />');
```

Comment:
why are we constructing HTML here instead of just using `ng-show` and setting variables in scope?